robot/client/client.go

```diff
@@ -786,6 +788,29 @@ func (rc *RobotClient) TransformPose(
 	return referenceframe.ProtobufToPoseInFrame(resp.Pose), nil
 }
 
+// TransformPointCloud will transform the pointcloud to the desired frame in the robot's frame system.
+// Do not move the robot between the generation of the initial pointcloud and the receipt
+// of the transformed pointcloud because that will make the transformations inaccurate.
+func (rc *RobotClient) TransformPointCloud(ctx context.Context, srcpc pointcloud.PointCloud, srcName, dstName string,
+) (pointcloud.PointCloud, error) {
+	// get the offset pose from a TransformPose request
+	sourceFrameZero := referenceframe.NewPoseInFrame(srcName, spatialmath.NewZeroPose())
+	transforms, err := referenceframe.LinkInFramesToTransformsProtobuf(nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := rc.client.TransformPose(ctx, &pb.TransformPoseRequest{
+		Destination:            dstName,
```

Comment:
Does `TransformPose` use the world frame if `Destination` is empty?

Reply:
changed default dst string to be World